### PR TITLE
Modify to follow Java8 syntax.

### PR DESCRIPTION
**Pull request by hotchemi**
## Overview

- Modify to use method reference at some point.
- Modify to use lambda expression at some point. 

---

I hereby agree to the terms of the JUnit Contributor License Agreement.

### CODE REVIEW

File: junit5-engine/src/main/java/org/junit/gen5/engine/junit5/discovery/JUnit5TestableFactory.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ JUnit5Testable fromMethod(Method testMethod, Class<?> clazz, String engineId) {
 	private JUnit5Testable createTestable(String uniqueId, String engineId, List<String> parts, JUnit5Testable last) {
 		if (parts.isEmpty())
 			return last;
-		JUnit5Testable next = null;
+		JUnit5Testable next;
```
Comment (hotchemi) [inline junit5-engine/src/main/java/org/junit/gen5/engine/junit5/discovery/JUnit5TestableFactory.java]: Just realized.

**Comment by sbrannen** [comment]
The build failed with our changes.

For future reference, please execute `gradle spotlessApply` before submitting a pull request.

**Comment by sbrannen** [comment]
Thanks for the pull request.

This has been merged into `master`.

**Comment by hotchemi** [comment]
Oh sorry and Thx!

**Pull request closed without merging**